psycopg/psycopg/errors.py

```diff
@@ -18,19 +18,202 @@
 
 # Copyright (C) 2020 The Psycopg Team
 
-from typing import Any, Dict, Optional, Sequence, Tuple, Type, Union
+from dataclasses import dataclass, field, fields
+from typing import Any, Callable, Dict, List, NoReturn, Optional, Sequence, Tuple, Type
+from typing import Union, TYPE_CHECKING
 from typing_extensions import TypeAlias
 from asyncio import CancelledError
 
 from .pq.abc import PGconn, PGresult
 from .pq._enums import DiagnosticField
 from ._compat import TypeGuard
 
+if TYPE_CHECKING:
+    from .pq.misc import PGnotify, ConninfoOption
+
 ErrorInfo: TypeAlias = Union[None, PGresult, Dict[int, Optional[bytes]]]
 
 _sqlcodes: Dict[str, "Type[Error]"] = {}
 
 
+@dataclass
+class FinishedPGconn:
+    """Finished libpq connection.
+
+    Attributes are set from a real `~pscopg.pq.PGconn` but any operations will
+    raise an `~psycopg.OperationalError`.
+    """
+
+    info: List["ConninfoOption"] = field(default_factory=list)
+
+    db: bytes = b""
+    user: bytes = b""
+    password: bytes = b""
+    host: bytes = b""
+    hostaddr: bytes = b""
+    port: bytes = b""
+    tty: bytes = b""
+    options: bytes = b""
+    status: int = 1
+    transaction_status: int = 0
+    pipeline_status: int = 0
```

Comment:
For these we can use symbolic values. I would use the UNKNOWN transaction status. I'll leave you to decide if OFF or ABORTED is a better value for the pipeline status (I assume OFF is).

```suggestion
    status: int = pq.ConnectionStatus.BAD.value
    transaction_status: int = pq.TransactionStatus.UNKNOWN.value
    pipeline_status: int = pq.PipelineStatus.OFF.value
```

Reply:
Better indeed; done now.